lighthouse-viewer/test/viewer-test-pptr.js

```diff
@@ -59,80 +65,173 @@ describe('Lighthouse Viewer', function() {
+      function getErrors(elems, selectors) {
+        return elems.map(el => {
+          const audit = el.closest(selectors.audits);
+          const auditTitle = audit && audit.querySelector(selectors.titles);
+          return {
+            explanation: el.textContent,
+            title: auditTitle ? auditTitle.textContent : 'Audit title unvailable',
+          };
+        });
       }
-      expected = expected.map(audit => audit.id);
-      const elementIds = await getAuditElementsIds({category, selector: selectors.audits});
 
-      assert.deepStrictEqual(
-        elementIds.sort(),
-        expected.sort(),
-        `${category} does not have the identical audits`
-      );
-    }
+      const errorSelectors = '.lh-audit-explanation, .tooltip--error';
+      const auditErrors = await viewerPage.$$eval(errorSelectors, getErrors, selectors);
+      const errors = auditErrors.filter(item => item.explanation.includes('Audit error:'));
+      assert.deepStrictEqual(errors, [], 'Audit errors found within the report');
+    });
   });
 
-  it('should contain a filmstrip', async () => {
-    const filmstrip = await viewerPage.$('.lh-filmstrip');
+  describe('PSI', () => {
+    let onApiRequestInterception;
+    let apiRequestInterceptionResolve;
 
-    assert.ok(!!filmstrip, `filmstrip is not available`);
-  });
+    function onRequest(interceptedRequest) {
+      if (interceptedRequest.url().includes('https://www.googleapis.com')) {
+        apiRequestInterceptionResolve(interceptedRequest);
+      } else {
+        interceptedRequest.continue();
+      }
+    }
+
+    beforeAll(async () => {
+      await viewerPage.setRequestInterception(true);
+      viewerPage.on('request', onRequest);
+    });
+
+    afterAll(async () => {
+      viewerPage.off('request', onRequest);
+      await viewerPage.setRequestInterception(false);
+    });
+
+    beforeEach(() => {
+      onApiRequestInterception = new Promise(resolve => apiRequestInterceptionResolve = resolve);
+    });
 
-  it('should not have any unexpected audit errors', async () => {
-    function getErrors(elems, selectors) {
-      return elems.map(el => {
-        const audit = el.closest(selectors.audits);
-        const auditTitle = audit && audit.querySelector(selectors.titles);
-        return {
-          explanation: el.textContent,
-          title: auditTitle ? auditTitle.textContent : 'Audit title unvailable',
-        };
+    it('should call out to PSI with all categories by default', async () => {
+      const url = `${viewerUrl}?url=https://www.example.com`;
+      await viewerPage.goto(url);
+
+      // Intercept and respond with sample lhr.
+      const interceptedRequest = await onApiRequestInterception;
+      const interceptedUrl = new URL(interceptedRequest.url());
+      expect(interceptedUrl.origin + interceptedUrl.pathname).toEqual('https://www.googleapis.com/pagespeedonline/v5/runPagespeed');
+      expect({
+        url: interceptedUrl.searchParams.get('url'),
```

Comment:
this is a bit funny. let's just do two expect() statements. one for url, one for category.